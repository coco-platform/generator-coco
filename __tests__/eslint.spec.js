/**
 * @description - coco:eslint generator
 * @author - huang.jian <hjj491229492@hotmail.com>
 */

// packages
const path = require('path');
const fs = require('fs');
const yeomon = require('yeoman-test');
const assert = require('yeoman-assert');
const hbs = require('handlebars');

describe('coco:eslint render', () => {
  const template = fs.readFileSync(
    path.resolve(
      process.cwd(),
      'generators',
      'eslint',
      'templates',
      '.eslintrc.yml.hbs'
    ),
    'utf8'
  );
  const compile = hbs.compile(template);

  it('should render properly with babel parser', () => {
    const ctx = {
      babel: true,
    };

    expect(compile(ctx)).toMatchSnapshot();
  });

  it('should render properly without babel parser', () => {
    const ctx = {
      babel: false,
    };

    expect(compile(ctx)).toMatchSnapshot();
  });

  it('should render property context', () => {
    const ctx = {
      context: ['node', 'jest'],
    };

    expect(compile(ctx)).toMatchSnapshot();
  });

  it('should render property environment extension', () => {
    const ctx = {
      extends: ['eslint:recommended'],
    };

    expect(compile(ctx)).toMatchSnapshot();
  });
});

describe('coco:eslint generator', () => {
  it('should output eslint configuration', () => {
    const eslint = path.resolve(__dirname, '../generators/eslint');
    const prompts = {
      babel: false,
      context: ['node', 'jest'],
      environment: 'Web',
    };

    return yeomon
      .run(eslint)
      .withPrompts(prompts)
      .then(() => {
        assert.file(['.eslintrc.yml']);

        const configuration = fs.readFileSync('.eslintrc.yml', 'utf8');

        expect(configuration).toMatchSnapshot();
      });
  });
});
