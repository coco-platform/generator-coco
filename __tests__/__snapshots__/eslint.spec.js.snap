// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coco:eslint generator should output eslint configuration 1`] = `
"parserOptions:
  ecmaVersion: 2018
  sourceType: module
  ecmaFeatures:
    jsx: true
env:
  node: true
  jest: true
extends:
  - 'eslint:recommended'
  - 'airbnb-base'
  - 'prettier'
rules:
  indent:
    - error
    - 2
    - SwitchCase: 1
  linebreak-style:
    - error
    - unix
  quotes:
    - single
    - allowTemplateLiterals: true
  semi:
    - error
    - always
  arrow-parens:
    - error
    - always
  max-len:
    - error
    - code: 120
"
`;

exports[`coco:eslint generator should render properly with babel parser 1`] = `
"parser: babel-eslint
parserOptions:
  ecmaVersion: 2018
  sourceType: module
  ecmaFeatures:
    jsx: true
rules:
  indent:
    - error
    - 2
    - SwitchCase: 1
  linebreak-style:
    - error
    - unix
  quotes:
    - single
    - allowTemplateLiterals: true
  semi:
    - error
    - always
  arrow-parens:
    - error
    - always
  max-len:
    - error
    - code: 120
"
`;

exports[`coco:eslint generator should render properly without babel parser 1`] = `
"parserOptions:
  ecmaVersion: 2018
  sourceType: module
  ecmaFeatures:
    jsx: true
rules:
  indent:
    - error
    - 2
    - SwitchCase: 1
  linebreak-style:
    - error
    - unix
  quotes:
    - single
    - allowTemplateLiterals: true
  semi:
    - error
    - always
  arrow-parens:
    - error
    - always
  max-len:
    - error
    - code: 120
"
`;

exports[`coco:eslint generator should render property context 1`] = `
"parserOptions:
  ecmaVersion: 2018
  sourceType: module
  ecmaFeatures:
    jsx: true
env:
  node: true
  jest: true
rules:
  indent:
    - error
    - 2
    - SwitchCase: 1
  linebreak-style:
    - error
    - unix
  quotes:
    - single
    - allowTemplateLiterals: true
  semi:
    - error
    - always
  arrow-parens:
    - error
    - always
  max-len:
    - error
    - code: 120
"
`;

exports[`coco:eslint generator should render property environment extension 1`] = `
"parserOptions:
  ecmaVersion: 2018
  sourceType: module
  ecmaFeatures:
    jsx: true
extends:
  - 'eslint:recommended'
rules:
  indent:
    - error
    - 2
    - SwitchCase: 1
  linebreak-style:
    - error
    - unix
  quotes:
    - single
    - allowTemplateLiterals: true
  semi:
    - error
    - always
  arrow-parens:
    - error
    - always
  max-len:
    - error
    - code: 120
"
`;

exports[`coco:eslint render should render properly with babel parser 1`] = `
"parser: babel-eslint
parserOptions:
  ecmaVersion: 2018
  sourceType: module
  ecmaFeatures:
    jsx: true
rules:
  indent:
    - error
    - 2
    - SwitchCase: 1
  linebreak-style:
    - error
    - unix
  quotes:
    - single
    - allowTemplateLiterals: true
  semi:
    - error
    - always
  arrow-parens:
    - error
    - always
  max-len:
    - error
    - code: 120
"
`;

exports[`coco:eslint render should render properly without babel parser 1`] = `
"parserOptions:
  ecmaVersion: 2018
  sourceType: module
  ecmaFeatures:
    jsx: true
rules:
  indent:
    - error
    - 2
    - SwitchCase: 1
  linebreak-style:
    - error
    - unix
  quotes:
    - single
    - allowTemplateLiterals: true
  semi:
    - error
    - always
  arrow-parens:
    - error
    - always
  max-len:
    - error
    - code: 120
"
`;

exports[`coco:eslint render should render property context 1`] = `
"parserOptions:
  ecmaVersion: 2018
  sourceType: module
  ecmaFeatures:
    jsx: true
env:
  node: true
  jest: true
rules:
  indent:
    - error
    - 2
    - SwitchCase: 1
  linebreak-style:
    - error
    - unix
  quotes:
    - single
    - allowTemplateLiterals: true
  semi:
    - error
    - always
  arrow-parens:
    - error
    - always
  max-len:
    - error
    - code: 120
"
`;

exports[`coco:eslint render should render property environment extension 1`] = `
"parserOptions:
  ecmaVersion: 2018
  sourceType: module
  ecmaFeatures:
    jsx: true
extends:
  - 'eslint:recommended'
rules:
  indent:
    - error
    - 2
    - SwitchCase: 1
  linebreak-style:
    - error
    - unix
  quotes:
    - single
    - allowTemplateLiterals: true
  semi:
    - error
    - always
  arrow-parens:
    - error
    - always
  max-len:
    - error
    - code: 120
"
`;
